addons/test_mail/tests/test_mail_message.py

```diff
@@ -1,10 +1,74 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
+import json
+
+from odoo.addons.base.models.ir_mail_server import MailDeliveryException
 from odoo.addons.test_mail.tests.common import TestMailCommon
 from odoo.exceptions import UserError
 from odoo.tools import is_html_empty, mute_logger, formataddr
-from odoo.tests import tagged, users
+from odoo.tests import tagged, users, HttpCase
+
+
+@tagged('mail_message', 'mail_controller', 'post_install', '-at_install')
+class TestMessageHelpers(TestMailCommon, HttpCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.user_employee.notification_type = 'email'
+        cls.test_records_simple, _partners = cls._create_records_for_batch(
+            'mail.test.simple', 3,
+        )
+
+    def setUp(self):
+        super().setUp()
+        # cleanup db
+        self.env['mail.notification'].search([('author_id', '=', self.partner_admin.id)]).unlink()
+
+        # generate crashed notifications
+        with mute_logger('odoo.addons.mail.models.mail_mail'), self.mock_mail_gateway():
+            def _send_email(*args, **kwargs):
+                raise MailDeliveryException("Some exception")
+            self.send_email_mocked.side_effect = _send_email
+
+            for record in self.test_records_simple:
+                record.with_user(self.user_admin).message_post(
+                    body="Setup",
+                    message_type='comment',
+                    partner_ids=self.partner_employee.ids,
+                    subtype_id=self.env.ref('mail.mt_comment').id,
+                )
+
+        # In the mean time, some FK deletes the record where the message is
+        # # scheduled, skipping its unlink() override
+        self.env.cr.execute(
+            f"DELETE FROM {self.test_records_simple._table} WHERE id = %s", (self.test_records_simple[2].id,)
+        )
+        self.env.invalidate_all()
+
+    def test_notify_cancel_by_type(self):
+        self.env.invalidate_all()
+        self.test_records_simple.with_user(self.user_admin).notify_cancel_by_type('email')
```

Comment:
Can't we add a cheap assert just to check it broadly worked? (i.e. check notification is marked cancelled)